lib/Sema/CSSimplify.cpp

```diff
@@ -2175,6 +2175,66 @@ bool ConstraintSystem::repairFailures(
     }
     break;
   }
+  case ConstraintLocator::ApplyArgToParam: {
+    auto loc = getConstraintLocator(locator);
+    if (repairByInsertingExplicitCall(lhs, rhs))
+      return true;
+
+    if (lhs->getOptionalObjectType() && !rhs->getOptionalObjectType()) {
+      conversionsOrFixes.push_back(
+          ForceOptional::create(*this, lhs, lhs->getOptionalObjectType(),
+                                loc));
+    }
+
+    Expr *argAnchor = nullptr;
+    if (auto *AE = dyn_cast<ApplyExpr>(loc->getAnchor())) {
+      auto *args = AE->getArg();
+
+      if (auto *TE = dyn_cast<TupleExpr>(args))
+        argAnchor = TE->getElement(elt.getValue());
+      if (auto *PE = dyn_cast<ParenExpr>(args))
+        argAnchor = PE->getSubExpr();
+    }
+
+    if (!argAnchor)
+      break;
+
+    // Find the resolved overload related to this
+    auto resolvedOverload = getResolvedOverloadSets();
```

Comment:
Two suggestions here:

 - please extract logic related to finding something in resolvedOverloadSets into a method on ConstraintSystem e.g. `findSelectedOverloadFor(Expr *)`
- This whole block could be extracted into a separate function which returns a wrapper so it could be opened and processed here.

Reply:
I'll implement that method in the primary PR: https://github.com/apple/swift/pull/25507 and migrate this one to make use of it!

Reply:
I'd really like to get this into 5.1 piecemeal, so it'd be great to do it here and then rebase with the primary PR which might not get into 5.1.

Reply:
Ah, I think there's a bit of confusion. This pull request is a superset of the other one. If function arguments are the priority, I can remove the member part from this pull request though and then finish building this one out though.

Reply:
Ah okay, I'm just lost in the PRs sorry - cool then, let's do it there and rebase this one.